### PR TITLE
Convert eks/cluster to aws-teams and aws-sso

### PR DESCRIPTION
**Pull request by Nuru**
## what

- Convert `eks/cluster` to `aws-teams` 
- Add `aws-sso` support to `eks/cluster`
- Undo automatic allowance of `identity` `aws-sso` permission sets into account roles added in #567 

## why

- Keep in sync with other modules
- #567 is a silent privilege escalation and not needed to accomplish desired goals

### CODE REVIEW

File: modules/account-map/modules/iam-roles/outputs.tf [hunk 1]
```diff
@@ -41,6 +41,14 @@ output "global_stage_name" {
   description = "The `null-label` `stage` value for the organization management account (where the `account-map` state is stored)"
 }
 
+output "current_account_account_name" {
```
[the remark this answers is not in the document]
Comment (Nuru) [reply modules/account-map/modules/iam-roles/outputs.tf:44]: To maintain consistency with other outputs, like:
- `identity_account_account_name`
- `root_account_account_name`

File: modules/eks/actions-runner-controller/README.md [hunk 2]
```diff
@@ -264,35 +264,29 @@ HRA being successfully notified about it, so as a safety measure, the `capacityR
...
-HRA will scale down the pool as jobs finish, so there is little cost to setting a long duration.
+If it happens that the capacity reservation expires before the job is finished, the HRA will scale down the pool
+by 2 instead of 1: once because the capacity reservation expired, and once because the job finished. This will
+also cause starvation of waiting jobs, because the next in line will have its timeout timer started but will not
+actually start running because no runner is available. And if `minReplicas` is set to zero, the pool will scale down
+to zero before finishing all the jobs, leaving some waiting indefinitely. This is why it is important to set the
+`webhook_startup_timeout` to a time long enough to cover the full time a job may have to wait between the time it is
+queued and the time it finishes, assuming that the HRA scales up the pool by 1 and runs the job on the new runner.
+
+:::info
```
[the remark this answers is not in the document]
Comment (Nuru) [reply modules/eks/actions-runner-controller/README.md:275]: Yes, I agree with you, but @osterman wants the README's optimize for how they look on docs.cloudposse.com, not for how they look on GitHub.

File: modules/account-map/modules/roles-to-principals/outputs.tf [hunk 3]
```diff
@@ -1,6 +1,10 @@
 output "principals" {
   value       = local.principals
-  description = "List of AWS principals corresponding to given input `role_map`"
+  description = "Consolidated list of AWS principals corresponding to given input `role_map`"
+}
+output "principals_map" {
```
Comment (Nuru) [inline modules/account-map/modules/roles-to-principals/outputs.tf:5]:
```suggestion
}

output "principals_map" {
```

File: modules/eks/actions-runner-controller/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (osterman) [reply modules/eks/actions-runner-controller/README.md:275]: Yes, the component documentation should be optimized for our public documentation. This is not uncommon from documentation found in other open source projects.

**Pull request merged**